blockchain_storage/src/store.rs

```diff
@@ -197,8 +197,9 @@ impl<Tx: for<'a> traits::GetMapRef<'a, Schema>> BlockchainStorageRead for StoreT
             Ok(Some(block)) => {
                 let begin = tx_index.get_byte_offset_in_block() as usize;
                 let end = begin + tx_index.get_serialized_size() as usize;
-                let tx = block.get(begin..end).expect("Transaction outside of block range");
-                let tx = Transaction::decode_all(tx).expect("Invalid tx encoding in DB");
+                let mut tx = block.get(begin..end).expect("Transaction outside of block range");
+                let decode_ptr = &mut tx;
+                let tx = Transaction::decode_all(decode_ptr).expect("Invalid tx encoding in DB");
```

Comment:
Why the temporary ptr? The `tx` variable is immediately shadowed anyway so this could be just:
```rust
let mut tx = block.get(begin..end).expect("Transaction outside of block range");
let tx = Transaction::decode_all(&mut tx).expect("Invalid tx encoding in DB");
```
(maybe rename the first `tx` to `encoded_tx` or something like that for clarity)

Reply:
`&mut tx` also doens't work. The function needs a mutable pointer, not a mutable object.